Code/Framework/AzToolsFramework/AzToolsFramework/Logger/TraceLogger.cpp

```diff
@@ -73,15 +73,15 @@ namespace AzToolsFramework
         AZStd::string logPath;
         StringFunc::Path::Join(logDirectory.c_str(), logFileName.c_str(), logPath);
 
-        m_logFile.reset(aznew LogFile(logPath.c_str()));
+        m_logFile.reset(aznew LogFile(logPath.c_str(), true));
         if (m_logFile)
         {
             m_logFile->SetMachineReadable(false);
             for (const LogMessage& message : m_startupLogSink)
             {
                 m_logFile->AppendLog(LogFile::SEV_NORMAL, message.window.c_str(), message.message.c_str());
             }
-            m_startupLogSink = {};
+            m_startupLogSink.clear();
```

Comment:
> 

This is in AzToolsFramework and only used within the Editor, so there is no need to clar the log during the first game tick.

Reply:
In the original implementation that used vectors, shrink_to_fit would not be necessary because the vector was being replaced with an empty vector, which would deallocate its content.
Switching to the list was supposed to be an optimization to avoid reallocating the vector and all of the strings inside whenever the reserve was exceeded.
Normally, I would reserve space with a vector but the number of lines is unknown from one application or run to the next. That could also be configurable if it continued to use a vector.
The cases that prompted this had pages and pages of warnings and errors from different systems that have mostly been addressed by me or their maintainers.

Reply:
> This is in AzToolsFramework and only used within the Editor, so there is no need to clear the log during the first game tick.

Maybe some other point then, the takeaway is to clear `m_startupLogSink` after the core has been bootstrapped and the log file should have been assigned.

>Normally, I would reserve space with a vector but the number of lines is unknown from one application or run to the next. That could also be configurable if it continued to use a vector.

It would still be beneficial to reserve a reasonably large number upfront to reduce the number of times the buffer needs to be resized. It can still happen a few times, but that's still better than the current situation. (In fact I would be more concerned with the number of `AZStd::string` allocations.) It's fine in this case to overshoot it as well because `m_startupLogSink` only holds memory during a period when there should be plenty, given that code is added to free that memory early on as well.